File: server/src/main/java/com/ps/server/dto/CreateMoveDTO.java
package com.ps.server.dto;

import com.ps.server.Logic.Position;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class CreateMoveDTO {
    private Long playerId;
    private Position origin;
    private Position destination;

    public Long getPlayerId() {
        return playerId;
    }

    public Position getOrigin() {
        return origin;
    }

    public Position getDestination() {
        return destination;
    }

    public void setPlayerId(Long playerId) {
        this.playerId = playerId;
    }

    public void setOrigin(Position origin) {
        this.origin = origin;
    }

    public void setDestination(Position destination) {
        this.destination = destination;
    }
}

File: client/src/app/DTO/createMoveDTO.ts
import {PositionDTO} from "./positionDTO";


/** @ignore*/
export class CreateMoveDTO {
    /** */
    public playerId: number;
    /** */
    public origin: PositionDTO;
    /** */
    public destination: PositionDTO;

    /** @ignore */
    constructor(playerId: number, origin: PositionDTO, destination: PositionDTO) {
        this.playerId = playerId;
        this.origin = origin;
        this.destination = destination;
    }
}